src/components/Calculator.js

```diff
@@ -19,122 +18,86 @@ const buttonStyle = (bgColor, w) => ({
-        <Button label="+" bgColor="orange" setValue={this.setValue} />
-        <br />
-        <Button label={0} setValue={this.setValue} width="200px" />
-        <Button label="." setValue={this.setValue} />
-        <Button label="=" bgColor="orange" setValue={this.setValue} />
-      </div>
-    );
-  }
-}
-
-class Input extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
-
-  render() {
-    const { value } = this.props;
-    return <div id="input" type="number" placeholder="Input">{ value }</div>;
-  }
-}
+  return (
+    <div>
+      <Display value={value} />
+      <Button label="AC" updateValue={() => updateValue('AC')} />
+      <Button label="DEL" updateValue={() => updateValue('DEL')} />
+      <Button label="%" updateValue={() => updateValue('%')} />
+      <Button label="/" bgColor="orange" updateValue={() => updateValue('/')} />
+      <br />
+      <Button label="7" updateValue={() => updateValue('7')} />
+      <Button label="8" updateValue={() => updateValue('8')} />
+      <Button label="9" updateValue={() => updateValue('9')} />
+      <Button label="*" bgColor="orange" updateValue={() => updateValue('*')} />
+      <br />
+      <Button label="4" updateValue={() => updateValue('4')} />
+      <Button label="5" updateValue={() => updateValue('5')} />
+      <Button label="6" updateValue={() => updateValue('6')} />
+      <Button label="-" bgColor="orange" updateValue={() => updateValue('-')} />
+      <br />
+      <Button label="1" updateValue={() => updateValue('1')} />
+      <Button label="2" updateValue={() => updateValue('2')} />
+      <Button label="3" updateValue={() => updateValue('3')} />
+      <Button label="+" bgColor="orange" updateValue={() => updateValue('+')} />
+      <br />
+      <Button label="0" updateValue={() => updateValue('0')} width="200px" />
+      <Button label="." updateValue={() => updateValue('.')} />
+      <Button label="=" bgColor="orange" updateValue={() => updateValue('=')} />
+    </div>
+  );
+};
 
-Input.propTypes = {
-  value: PropTypes.number.isRequired,
+const Display = (props) => {
+  const { value } = props;
+  return <div id="display" type="number">{ value }</div>;
 };
 
-class Button extends React.Component {
-  constructor(props) {
-    super(props);
-    this.setValue = this.setValue.bind(this);
-    this.state = {};
-  }
+Display.propTypes = {
+  value: PropTypes.string.isRequired,
+};
 
-  setValue(e) {
-    const { setValue } = this.props;
-    setValue(e.target.value);
-  }
+const Button = (props) => {
+  const {
+    label,
+    bgColor,
+    width,
+    updateValue,
+  } = props;
 
-  render() {
-    const { label, bgColor, width } = this.props;
-    return <button style={buttonStyle(bgColor, width)} type="button" onClick={this.setValue} value={label}>{ label }</button>;
-  }
-}
+  return <button style={buttonStyle(bgColor, width)} type="button" onClick={() => updateValue()}>{ label }</button>;
+};
```

Comment:
- [ ] it would be better if you move the button to separate file cause button it's component itself